### PR TITLE
com.github.marktext.marktext

### PR DESCRIPTION
**Pull request by fxha**

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.github.marktext.marktext

**Comment by fxha** [comment]
@flathub Do I have to do something else or could you please review the PR?

File: data/com.github.marktext.marktext.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    </description>
+    <url type="homepage">https://marktext.github.io/website/</url>
+    <url type="bugtracker">https://github.com/marktext/marktext/issues</url>
+    <screenshots>
+        <!-- TODO: Update links after merge -->
+        <screenshot type="default">
+            <caption>Main</caption>
+            <image>https://raw.githubusercontent.com/marktext/website/master/docs/img/light.57e00131.jpg</image>
+        </screenshot>
+        <!--<screenshot type="default">
```
Comment (nedrichards) [inline data/com.github.marktext.marktext.appdata.xml]: If you're not using these screenshots then please remove them from the file.

**Comment by nedrichards** [comment]
@fxha do you plan to send the desktop file and appdata upstream? That would be most helpful.

**Comment by fxha** [comment]
@nedrichards Yes, that's true. Then I'll update the screenshots while I'm at it too.

**Comment by nedrichards** [comment]
And apologies for the delay - (personally I've been travelling for work so I'm very behind on flathub volunteer time)

**Comment by nedrichards** [comment]
bot, build com.github.marktext.marktext

**Comment by nedrichards** [comment]
I have questions about the general approach. Is there a reason you can't build this from source? There's a package-lock.json so an approach like https://github.com/flatpak/flatpak-builder-tools/tree/master/npm should work (unless there's some tricksy binary dependencies). As well as being a bit better in terms of reproducibility this would also let us target at least i386 and arm (and aarch64 soon, when they upgrade to the latest electron)

**Comment by fxha** [comment]
I thought that you cannot download files on the build process. How do I download the packages before `flatpak-builder` command? Or do I need to run `flatpak-npm-generator.py` and link `generated-sources.json` manually? And are there known problems with `electron-builder` because we use it to build Mark Text and it use pre-build electron binaries, which are downloaded during build process. Perhaps we need [hunspell](https://hunspell.github.io/) library for spell checking.

> this would also let us target at least i386 and arm (and aarch64 soon, when they upgrade to the latest electron)

I don't know anyone using x86 desktop (GUI) Linux. We currently support only x86_64 and x86 on Windows and no support for ARM is planned. x86 build will be okay if build from source.

> Note that this requires io.atom.electron.BaseApp installed.

Do I need the atom package or is `org.electronjs.Electron2.BaseApp` sufficient?

**Comment by nedrichards** [comment]
You run the npm generator and link in the generated-sources.json manually. That handles downloading the files during the 'sources download' phase and making sure they're available in a local cache for npm and yarn.

For libraries that you need as dependencies (as opposed to binaries downloaded by npm) you could just add them as normal modules, that's basically what the baseapp is for - adding modules that essentially every electron app needs.

If you're using Electron 1.x then you need io.atom.electron.BaseApp if you're using 2.x then org.electronjs.Electron2.BaseApp (the big difference on Linux is that they drop support for GTK2 in 2.x so the build is a lot smaller).

All of these reasons not to use the tools right now seem legit - so I'm going to merge this as is and I'd recommend that you experiment with the other toolchain in a PR. (as an aside, it really would be great if someone resuscitated the flatpak support in electron builder/bundler/that whole ecosystem for projects like this)

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.marktext.marktext

**Comment by fxha** [comment]
@nedrichards Thanks! Could you please give @Jocs access to the repository too.

I'm a bit busy right now but I'll will look into it.

**Comment by nedrichards** [comment]
@fxha added.

**Comment by Jocs** [comment]
@fxha @nedrichards thanks

**Pull request closed without merging**